apisix/plugins/real-ip.lua

```diff
@@ -84,6 +104,22 @@ local function get_addr(conf, ctx)
             return addrs
         end
 
+        if conf.recursive and conf.trusted_addresses then
+            local split_addrs = {}
+            local match_itr = str_gmatch(addrs, "[^,%s*]+")
+            for itr in match_itr do
+                tb_insert(split_addrs, itr)
+            end
+
+            for i = #split_addrs, 1, -1 do
```

Comment:
```suggestion
            for i = #split_addrs, 2, -1 do
```
We don't care if the first one can be trusted.